### ISSUE TITLE
Is knowledge really transferred?

### ISSUE THREAD

**Issue by igasparini** [CGuangyan-BIT/PointGPT#33]
Hi, thank you for your work and the codebase :)

My question relates to how knowledge from a pre-trained PointGPT transfers to Point Transformer during post-pre-training. We load the weights and we add the CLS token and the classification layers. But the sequence length and attention mask shape for the extractor in post-pre-training is different from the pre-training, because of the introduced CLS token. So it seems to me that the knowledge in the extractor is wasted because of misaligned sequences and therefore wrong positional relationships, and the transformer has to re-learn from scratch. Or am I understanding it wrong?

Thank you for your feedback

**Comment by igasparini** [comment]
After looking into it, this is indeed common practice and although absolute positional relationships get disrupted, feature representations and relative attention patterns should remain preserved. This is also why fine-tuning with 128 groups vs 64 groups of the pre-training (2048 vs 1024 points in the point clouds) works.
